src/Elements/Spatial/Grid2d.cs

```diff
@@ -134,6 +138,11 @@ public Grid2d(IList<Polygon> boundaries, Transform transform = null)
             toGrid.Invert();
 
             var transformedBoundaries = toGrid.OfPolygons(boundaries);
+            var allVerticesZ = transformedBoundaries.SelectMany(b => b.Vertices).Select(v => v.Z);
+            if(allVerticesZ.Any(z => !z.ApproximatelyEquals(0))) {
+                throw new Exception("After transform, this polygon was not in the XY Plane. Please ensure that all your geometry as well as any provided transform all lie in the same plane.");
```

Comment:
When I throw an exception inside a constructor, I like to say something like:
> The Grid2d could not be constructed. After transform, the polygon was not in the XY plane...

This enables the developer to read the exception method an known exactly where the exception was thrown without having to read the stack trace. It's also nice for our developers who are less familiar with stack traces and would just like the computer to tell them exactly what went wrong in human terms.